Library/src/Multiplayer/Components/TextSpaceComponent.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "CSP/Multiplayer/Components/TextSpaceComponent.h"
+
+#include "Debug/Logging.h"
+#include "Memory/Memory.h"
+#include "Multiplayer/Script/ComponentBinding/TextSpaceComponentScriptInterface.h"
+
+
+namespace csp::multiplayer
+{
+
+TextSpaceComponent::TextSpaceComponent(SpaceEntity* Parent) : ComponentBase(ComponentType::Image, Parent)
+{
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Position)]			 = csp::common::Vector3::Zero();
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Rotation)]			 = csp::common::Vector4 {0, 0, 0, 1};
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Scale)]				 = csp::common::Vector3::One();
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Text)]				 = "";
+	Properties[static_cast<uint32_t>(TextPropertyKeys::TextColor)]			 = csp::common::Vector3::Zero();
+	Properties[static_cast<uint32_t>(TextPropertyKeys::BackgroundColor)]	 = csp::common::Vector3::Zero();
+	Properties[static_cast<uint32_t>(TextPropertyKeys::IsBackgroundVisible)] = true;
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Width)]				 = 1.0f;
+	Properties[static_cast<uint32_t>(TextPropertyKeys::Height)]				 = 1.0f;
+	Properties[static_cast<uint32_t>(TextPropertyKeys::BillboardMode)]		 = static_cast<int64_t>(BillboardMode::Off);
+	Properties[static_cast<uint32_t>(TextPropertyKeys::IsVisible)]			 = false;
```

Comment:
I think this should be visible by default ? Unless I am misunderstanding and these values are not defaults.

Reply:
No you're right, visible should be true by default, as should visible in AR.

Reply:
Addressed in latest (opened a separate conversation for the AR one) 👍